waku/node/waku_node.nim

```diff
@@ -482,35 +581,86 @@ proc filterUnsubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], conten
     for pubsub, topics in topicMap.pairs:
       info "unsubscribed from topic", pubsubTopic=pubsub, contentTopics=topics
 
-# TODO: Move to application module (e.g., wakunode2.nim)
-proc subscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentTopics: ContentTopic|seq[ContentTopic], handler: FilterPushHandler) {.async, gcsafe,
-  deprecated: "Use the explicit destination peer procedure. Use 'node.filterSubscribe()' instead.".} =
-  ## Registers for messages that match a specific filter. Triggers the handler whenever a message is received.
-  if node.wakuFilterClientLegacy.isNil():
-    error "cannot register filter subscription to topic", error="waku filter client is nil"
-    return
+proc filterUnsubscribe*(node: WakuNode,
+                          pubsubTopic: Option[PubsubTopic],
+                          contentTopics: Option[seq[ContentTopic]],
```

Comment:
I don't think this should be an `Option`. `seq[ContentTopic]` can be empty, but that would result in an `err` returned by the underlying client protocol. We don't want to build much logic into `waku_node` belonging to underlying protocols - it simply exists to mount, integrate and expose a common API for underlying protocols. The exception here is that we do provide autosharding "middleware" in the `waku_node` - in other words, logic to populate the `pubsubTopic` if absent from API calls. But even this should probably be extracted to another module. The `waku_node` should in theory just provide a `filterSubscribe`, `filterUnsubscribe` and `filterUnsubscribeAll`- not interpret certain API calls as one or the other based on added logic.